lib/dbadaptor.py

```diff
@@ -196,14 +197,14 @@ def create_table(self, conn, name, columns, keys, recreate):
                 for ky in auto_keys:
                     if ky in prim_keys and columns[ky] != 'INTEGER':
                         columns[ky] = 'INTEGER'
-        super(SQLDatabaseAdaptor, self).create_table(conn, name, columns, keys,
-                                                     recreate)
+        super(SQLDatabaseAdaptor, self).create_table(conn, name, columns, keys, recreate)
 
     def fetch_tables(self, conn):
         '''Fetch all the table names in the database'''
-        c = conn.cursor()
-        c.execute("SELECT name FROM sqlite_master WHERE type='table'")
-        return list(c)
+        with closing(conn.cursor()) as c:
+            c.execute("SELECT name FROM sqlite_master WHERE type='table'")
+            out = c.fetchall()
```

Comment:
Actually, `list` will get the results automatically. 
`c.fetchall()` is not necessary, but it's ok, we can keep it. It's clearer

Reply:
When using context managers, c is closed outside of the with statement, so list(c) would not work.